File: Object/C++/animals/dog.cpp
#include "dog.h"

Dog::Dog(std::string myName, std::string ownName): Animal(++number, myName, ownName) {}
Dog::Dog(std::string myName): Animal(++number, myName) {}

int Dog::DogCount() {
    return number;
}

void Dog::Identify() const {
    printf("A dog!\n");
    Animal::Identify();
}

void Dog::MakeNoise() const {
    printf("Woof!\n");
}

File: Object/Java/animals/Dog.java
public class Dog extends Animal {
    private static int number = 0;

    public Dog(String anName) {
        super(++number, anName);
    }

    public Dog(String anName, String anOwner) {
        super(++number, anName, anOwner);
    }

    public void MakeNoise() {
        System.out.println("Woof!");
    }

    public void Identify() {
        System.out.println("I am a dog!");
        super.Identify();
    }

    public static int DogCount() {
        return number;
    }
}
